src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration.Tests/samples/Sample7_ConfiguringPipeline.cs

```diff
@@ -20,29 +20,26 @@ public partial class ConfigurationSamples
         public async Task ConfiguringPipeline()
         {
             // this instance will hold pipeline creation options
-            var options = ConfigurationClient.CreateDefaultPipelineOptions();
+            var options = new HttpPipelineOptions();
 
             // specify custon HttpClient
-            options.Transport = new HttpClientTransport(s_client);
-
-            // remove logging policy
-            options.LoggingPolicy = null; 
+            options.UseHttpClient(s_client);
 
             // specify custom retry policy options
-            options.RetryPolicy = RetryPolicy.CreateFixed(
+            options.AddFixedRetry(
                 maxRetries: 10,
                 delay: TimeSpan.FromSeconds(1),
                 retriableCodes: new int[] {
-                    500, // Internal Server Error 
+                    500, // Internal Server Error
                     504  // Gateway Timeout
                 }
             );
 
             // add a policy (custom behavior) that executes once per client call
-            options.AddPerCallPolicy(new AddHeaderPolicy());
+            options.Append(HttpPipelineSection.PreRetry, next => new AddHeaderPolicy(next));
```

Comment:
I really dislike the delegates. Also, I find "section" terminology to be strange. Maybe we should do:
```c#
public enum PolicyRunner {
    OncePerCall,
    OncePerRetry,
}
```
or something like that.

Reply:
Delegates also align us more with HttpMessageHandler if we decide to go that direction.

I added the `Retry` section that allows you to have multiple retry policies and to override all of them at once if you want to.

See https://github.com/Azure/azure-sdk-for-net/pull/5524/files#diff-97159a9a3212cad4cb8778aeb3b026fcR44